### PR TITLE
enable node support

### PR DESCRIPTION
**Pull request by wangleis**
### Details:
 - *enable nodes support on Linux and add test cases*
 - *enable nodes support on Windows which is same as sockets*
 - *update base functions to support latency mode per numa node, per socket, and per platform. Default setting in this PR is latency mode per socket which is same as previous behavior*

### Tickets:
 - *ticket-id*

### CODE REVIEW

**Review by sunxiaoxia2022 — APPROVED** [review]

File: src/inference/src/streams_executor.hpp [hunk 1]
```diff
@@ -34,35 +35,58 @@ class CPU {
...
+/**
+ * @brief      Parse nodes infomation to update _sockets, proc_type_table and cpu_mapping_table on Linux
+ * @param[in]  node_info_table nodes information for this platform.
+ * @param[in]  _nodes total number for nodes in system
+ * @param[out] _sockets total number for sockets in system
+ * @param[out] _proc_type_table summary table of number of processors per type
+ * @param[out] _cpu_mapping_table CPU mapping table for each processor
+ * @return
+ */
+void parse_node_info_linux(const std::vector<std::string> node_info_table,
```
Comment (dmitry-gorokhov) [inline src/inference/src/streams_executor.hpp:47]: why some arguments have `_` prefix and some don't?

File: src/inference/tests/unit/cpu_map_parser.cpp
Comment (dmitry-gorokhov) [inline src/inference/tests/unit/cpu_map_parser.cpp:1]: This test file becomes to big. Lets create separate folder /unit/cpu_map_parser and split the tests on separate cpps based on parsing type.

File: src/inference/src/os/lin/lin_system_conf.cpp [hunk 2]
```diff
@@ -516,28 +623,35 @@ void parse_freq_info_linux(const std::vector<std::vector<std::string>> system_in
...
+    _sockets = (_sockets > 0) ? _sockets + 1 : 1;
 
-    if (_sockets >= 1) {
-        _proc_type_table.resize(_sockets + 2, std::vector<int>(PROC_TYPE_TABLE_SIZE, 0));
-        for (int n = 0; n < _processors; n++) {
-            _proc_type_table[0][ALL_PROC]++;
-            _proc_type_table[_cpu_mapping_table[n][CPU_MAP_SOCKET_ID] + 1][ALL_PROC]++;
+    if (node_info_table.size() == 0) {
+        if ((_sockets > 1) && (ecore_enabled)) {
+            _sockets = 1;
```
Comment (dmitry-gorokhov) [inline src/inference/src/os/lin/lin_system_conf.cpp:630]: It looks not obvious. Could you please clarify why we consider hybrid system to have single socket/numa_node?

File: src/inference/src/streams_executor.hpp [hunk 3]
```diff
@@ -34,35 +35,58 @@ class CPU {
 CPU& cpu_info();
 
 #ifdef __linux__
+/**
+ * @brief      Parse nodes infomation to update _sockets, proc_type_table and cpu_mapping_table on Linux
```
[the remark this answers is not in the document]
Comment (wangleis) [reply src/inference/src/streams_executor.hpp:39]: updated

File: src/inference/tests/unit/cpu_map_parser.cpp
Comment (wangleis) [reply src/inference/tests/unit/cpu_map_parser.cpp:1]: updated

File: src/inference/tests/unit/cpu_map_parser.cpp [hunk 4]
```diff
@@ -289,111 +296,447 @@ LinuxCpuMapTestCase cache_2sockets_104cores_hyperthreading = {
...
+        {"30,78", "30,78", "24-47,72-95"}, {"31,79", "31,79", "24-47,72-95"}, {"32,80", "32,80", "24-47,72-95"},
+        {"33,81", "33,81", "24-47,72-95"}, {"34,82", "34,82", "24-47,72-95"}, {"35,83", "35,83", "24-47,72-95"},
+        {"36,84", "36,84", "24-47,72-95"}, {"37,85", "37,85", "24-47,72-95"}, {"38,86", "38,86", "24-47,72-95"},
+        {"39,87", "39,87", "24-47,72-95"}, {"40,88", "40,88", "24-47,72-95"}, {"41,89", "41,89", "24-47,72-95"},
+        {"42,90", "42,90", "24-47,72-95"}, {"43,91", "43,91", "24-47,72-95"}, {"44,92", "44,92", "24-47,72-95"},
+        {"45,93", "45,93", "24-47,72-95"}, {"46,94", "46,94", "24-47,72-95"}, {"47,95", "47,95", "24-47,72-95"},
+    },
+    {{"0-23,48-71"}, {"24-47,72-95"}},
+};
+LinuxCpuMapTestCase cache_2sockets_48cores_hyperthreading_1 = {
```
[the remark this answers is not in the document]
Comment (wangleis) [reply src/inference/tests/unit/cpu_map_parser.cpp]: updated

File: src/inference/src/os/lin/lin_system_conf.cpp (hunk 2 above)
Comment (dmitry-gorokhov) [reply src/inference/src/os/lin/lin_system_conf.cpp:630]: Then such WAs should be described using comments. Otherwise it looks really confusing.

Comment (wangleis) [reply src/inference/src/os/lin/lin_system_conf.cpp]: updated

**Review by dmitry-gorokhov — APPROVED** [review]

**Comment by wangleis** [comment]
@dmitry-gorokhov enable base functions for latency stream per numa_node, per socket, or per platform. Please review again. Thanks.

File: src/inference/src/streams_executor.hpp
Comment (dmitry-gorokhov) [inline src/inference/src/streams_executor.hpp:1]: BTW just noticed that this filename has nothing common with sreams_executor. I would suggest to rename properly and put nearby `src/inference/src/os/` there it is actually used.

File: src/plugins/intel_cpu/src/config.h [hunk 5]
```diff
@@ -20,6 +20,12 @@
 namespace ov {
 namespace intel_cpu {
 
+enum class LatencyCandidateScope {
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/config.h:23]: 1. Lets to declaration inside `Config` structure.
2. I would propose to rename -> `LatencyThreadingMode`
3. Is the same logic extension is expected for Throughput mode?

File: src/plugins/intel_cpu/tests/unit/streams_info_table_test.cpp [hunk 6]
```diff
@@ -390,924 +393,1568 @@ class StreamsCalculationTests : public CommonTestUtils::TestsCommon,
...
+                                                  test_data.input_perf_hint,
+                                                  test_data.scopeOflatencyCandidate,
                                                   test_data.proc_type_table);
 
         ASSERT_EQ(test_data.stream_info_table, test_stream_info_table);
     }
 };
 
-StreamsCalculationTestCase _2sockets_104cores_latency_1 = {
+StreamsCalculationTestCase _2sockets_104cores_latency_platform_1 = {
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/tests/unit/streams_info_table_test.cpp:408]: Same ask as before: can we add corresponding commets to the test case?

File: src/plugins/intel_cpu/src/config.h [hunk 7]
```diff
@@ -59,6 +65,7 @@ struct Config {
     ov::hint::SchedulingCoreType schedulingCoreType = ov::hint::SchedulingCoreType::ANY_CORE;
     bool enableHyperThreading = true;
     bool changedHyperThreading = false;
+    LatencyCandidateScope scopeOflatencyCandidate = LatencyCandidateScope::PER_SOCKET;
```
[the remark this answers is not in the document]
Comment (wangleis) [reply src/plugins/intel_cpu/src/config.h:68]: If user explicitly set `nstreams=1` which is higher priority than LatencyThreadingMode (which will be hint in the future), we will create 1 stream with all CPU resource.

[the remark this answers is not in the document]
Comment (dmitry-gorokhov) [reply src/plugins/intel_cpu/src/config.h:68]: Got it. Lets directly state the priorities in the comments

File: src/plugins/intel_cpu/src/config.h (hunk 5 above)
Comment (wangleis) [reply src/plugins/intel_cpu/src/config.h:23]: Updated. 

Not yet. `LatencyThreadingMode` is defined for threads of one stream in Latency mode. We do not expect same configuration in both Latency mode and Throughput mode.

File: src/inference/src/streams_executor.hpp
Comment (wangleis) [reply src/inference/src/streams_executor.hpp:1]: updated

File: src/plugins/intel_cpu/tests/unit/streams_info_table_test.cpp (hunk 6 above)
Comment (wangleis) [reply src/plugins/intel_cpu/tests/unit/streams_info_table_test.cpp:408]: updated

File: src/plugins/intel_cpu/src/config.h (hunk 7 above)
[the remark this answers is not in the document]
Comment (wangleis) [reply src/plugins/intel_cpu/src/config.h]: updated

**Pull request merged**